http_client_wrapper.go

```diff
@@ -71,15 +80,28 @@ func (h httpClientWrapper) ExecuteRequest(path, method string, body []byte) (*ht
 			if err != nil {
 				return errors.Wrapf(err, "error reading failed request body")
 			}
-			errResp := errors.Errorf("received non success status code %d with body: %s", res.StatusCode, string(body))
+			errResp := BuildResponseError(res.StatusCode, errors.Errorf("received non success status code %d with body: %s", res.StatusCode, string(body)))
 			if res.StatusCode == 400 || res.StatusCode == 401 || res.StatusCode == 404 {
 				// unrecoverable since probably bad payload or n
-				return retry.Unrecoverable(errResp)
+				return retry.Unrecoverable(BuildResponseError(res.StatusCode, errResp))
```

Comment:
**suggestion (non-blocking):** Wrapping this twice seems strange to me. What if we did something like
```
type HTTPResponseError struct {
	StatusCode int
	Body       []byte
}

func (e *HTTPResponseError) Error() string {
	return fmt.Sprintf("status code %d with body %s", e.StatusCode, e.Body)
}
```
created that, and wrapped if the status code indicates unrecoverable?

Reply:
**thought:** It might also help the discussion here to talk about how you'd want to use this new behavior of `HTTPClientWrapper` to pass along 404s. Right now it seems like you have to search for the string `404`.